cluster-agent/controllers/argoproj.io/namespace_reconciler.go

```diff
@@ -269,21 +269,26 @@ func compareApplications(applicationFromArgoCD appv1.Application, applicationFro
 	}
 
 	var isAutomatedPruneUpdateNeeded bool
-	if applicationFromArgoCD.Spec.SyncPolicy.Automated.Prune != applicationFromDB.Spec.SyncPolicy.Automated.Prune {
-		log.Info("Prune field in ArgoCD and DB entry is not in Sync.")
-		isAutomatedPruneUpdateNeeded = true
-	}
-
 	var isAutomatedSelfHealUpdateNeeded bool
-	if applicationFromArgoCD.Spec.SyncPolicy.Automated.SelfHeal != applicationFromDB.Spec.SyncPolicy.Automated.SelfHeal {
-		log.Info("SelfHeal field in ArgoCD and DB entry is not in Sync.")
-		isAutomatedSelfHealUpdateNeeded = true
-	}
-
 	var isAutomatedAllowEmptyUpdateNeeded bool
-	if applicationFromArgoCD.Spec.SyncPolicy.Automated.AllowEmpty != applicationFromDB.Spec.SyncPolicy.Automated.AllowEmpty {
-		log.Info("AllowEmpty field in ArgoCD and DB entry is not in Sync.")
-		isAutomatedAllowEmptyUpdateNeeded = true
+
```

Comment:
I think this will miss the case where, for example:
- syncPolicy is nil in the database
- syncPolicy is not-nil on the cluster
- or vice versa

E.g. we should handle the case where at least one of them is nil.

With the new version of the logic, it will return false, when it should return true.

It also could not hurt to add some additional unit tests for this function to catch this.

Reply:
Logic is changed now, PTAL.